packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js

```diff
@@ -42,7 +49,19 @@ function aggregateDirectory(languageMessages, currentDirectory, inputFileSystem)
   } catch (e) {
     // not outputting anything here as the catching of the directory not existing is not an error in this case
   }
+}
 
+// Aggregate all translation messages under a custom directory
+function aggregateCustomDirectory(languageMessages, rootDir, inputFileSystem) {
```

Comment:
This function is written quite differently than our current setup. I'd prefer it follows the aggregation setup we have for consistency. Additionally, this seems to assume we would provided the root dir of the translations directory, but I would assume the translation directory path would be the path provided.

 I would expect something like
```
function aggregateCustomDirectory(languageMessages, customTranslationsDirectory, inputFileSystem) {
  // Check the directory for translations
  readTranslations(languageMessages, customTranslationsDirectory, inputFileSystem);   
  // Check sub directories for translations
    try {
    getDirectories(customTranslationsDirectory, inputFileSystem).forEach((subDirectory) => {
      aggregateDirectory(languageMessages, path.resolve(customTranslationsDirectory, subDirectory), inputFileSystem);
    });
  } catch (e) {
    // not outputting anything here as the catching of the directories not existing is not an error in this case
  }

  return languageMessages;
}
```

Reply:
This makes me think we could possibly just reuse `aggregateDirectory` and update [this line]( https://github.com/cerner/terra-core/blob/6b248786e32c373ff25e03fba693855d2fb1ef7f/packages/terra-i18n-plugin/src/I18nAggregatorPlugin.js#L66) to check if is customDir

Reply:
> This function is written quite differently than our current setup. I'd prefer it follows the aggregation setup we have for consistency. Additionally, this seems to assume we would provided the root dir of the translations directory, but I would assume the translation directory path would be the path provided.

This function does two things:
- Find all paths to translation folders under the custom directory. Updated the param `rootDir` to `customTranslationsDirectory` to avoid ambiguous. 
- Iterate the array of target paths and read translation messages from them

The current setup reads the translation messages when searching the target folder recursively. I was trying to reuse the same setup. But the searching pattern is a bit different:
- The current algorithm assumes the searching path follows a specific directory structure, for example:
   - node_modules
      - packages 1
        - node_modules
        - folder 11
        - folder 12
        - translations
      - packages 2
        - node_modules
        - folder 21
        - folder 22
        - translations


 The search paths include:
  - node_modules/packages1
  - node_modules/packages1/node_modules...
  - node_modules/packages2
  - node_modules/packages2/node_modules...

It is very efficient since it doesn't go into folder 11/12/21/22. 

For the custom directory searching, it will look for all possible paths, e.g.:
   - customDir
      - folder 1
        - folder 11
        - folder 12
        - translations
      - folder 2
        - folder 21
        - folder 22
        - translations
     - translations

 The search paths include:
  - customDir
  - customDir/folder 1/
  - customDir/folder 1/folder 11
  - customDir/folder 1/folder 12
  - customDir/folder 1/translations
  - customDir/folder 2/
  - customDir/folder 2/folder 21
  - customDir/folder 2/folder 22
  - customDir/folder 2/translations
  - customDir/translations

I prefer to decouple the path searching and files reading, which make it easy to follow the code.

Reply:
Also since this option is an optional, decoupling it from node_modules searching would preserve the default performance if the custom directory is not specified

Reply:
With the suggestions I made, I would expect this structure to be handled by adding an additional bool param to be passed to aggregateDirectory() to indicate `searchNodeModules`. 
Then I'd expect this this line:

```
const nodeMoudlesPath = path.resolve(currentDirectory, 'node_modules');
```

to be updated to 

```js
const nodeMoudlesPath = searchNodeModules ? path.resolve(currentDirectory, 'node_modules') : path.resolve(currentDirectory);
```

This should accomplish the folder structor you have indicated.